src/modules/deposit/screens/DepositConfirmation/index.tsx

```diff
@@ -78,12 +80,48 @@ function DepositConfirmation({
     liquidationThresholdAfter
   );
 
+  // Get approve and supply transactions without using permit flow
   const handleGetTransactions = async () => {
-    return lendingPool.deposit({
+    if (currentMarketData.v3) {
+      // TO-DO: No need for this cast once a single Pool type is used in use-tx-builder-context
+      const newPool: Pool = lendingPool as Pool;
+      return await newPool.supply({
+        user: user.id,
+        reserve: poolReserve.underlyingAsset,
+        amount: amount.toString(),
+        referralCode: undefined,
+      });
+    } else {
+      return await lendingPool.deposit({
+        user: user.id,
+        reserve: poolReserve.underlyingAsset,
+        amount: amount.toString(),
+        referralCode: getReferralCode(),
+      });
+    }
+  };
+
+  // Generate signature request payload
+  const handleGetPermitSignatureRequest = async () => {
+    // TO-DO: No need for this cast once a single Pool type is ued in use-tx-builder-context
+    const newPool: Pool = lendingPool as Pool;
+    return await newPool.signERC20Approval({
+      user: user.id,
+      reserve: poolReserve.underlyingAsset,
+      amount: amount.toString(),
+    });
+  };
+
+  // Generate supply transaction with signed permit
+  const handleGetPermitSupply = async (signature: string) => {
+    // TO-DO: No need for this cast once a single Pool type is ued in use-tx-builder-context
+    const newPool: Pool = lendingPool as Pool;
+    return await newPool.supplyWithPermit({
       user: user.id,
       reserve: poolReserve.underlyingAsset,
       amount: amount.toString(),
-      referralCode: getReferralCode(),
```

Comment:
you can leave this method, as if it doesnt find referralCode will return undefined, which is already correct